simulation/halsim_ws_client/src/main/native/cpp/HALSimWSClient.cpp

```diff
@@ -0,0 +1,197 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "HALSimWSClient.h"
+
+#include <wpi/SmallString.h>
+#include <wpi/raw_ostream.h>
+#include <wpi/uv/util.h>
+
+#include "HALSimWSClientConnection.h"
+
+#define MAX_TCP_CONNECT_ATTEMPTS 5
+#define TCP_CONNECT_ATTEMPT_TIMEOUT_MS 1000
```

Comment:
This is a style thing, but use `static constexpr int kFooBar = 1000;` instead of `#define`.